### DISCUSSION TITLE
unknown status code : 17028 ! I am trying to implement Phone Auth with flutterfire and I receive error saying "SMS verification code request failed: unknown status code: 17028 Invalid request fields in play_integrity_token"

### DISCUSSION THREAD

**Discussion by hyderali0889** [firebase/flutterfire#10785] [Q&A] [answered]
I've added the SHA-1 and SHA-256 to my firebase project and enabled the auth check with play integrity also checked the app name and SHA-1, SHA256 keys nothing seems to work and this error keeps on popping. I've redownloaded the google-service.json file also cleaned my project and recompiled it with every new update but nothing works.

My Code is

```
 void signInWithPhone() async {
      try {    
        FirebaseAuth auth = FirebaseAuth.instance;
        await auth.verifyPhoneNumber(
            forceResendingToken: token,
            phoneNumber: "+92${widget.phoneController.text.trim()}",
            verificationCompleted: (credential) async {
              await auth.signInWithCredential(credential);
              setState(() {
                hello = "Logged In";
              });
            },
            verificationFailed: (verificationFailed) {
              setState(() {
                hello = "Error Occured $verificationFailed";
              });
            },
            codeSent: (codeSent, otp) {
              setState(() {
                hello = "Code sent";
                token = otp;
              });
            },
            codeAutoRetrievalTimeout: (codeAutoRetrievalTimeout) {});
        widget.controller.startLoading(false);
      } catch (e) {
print(e);        
      }
    }
```

the error says 
> [SmsRetrieverHelper] SMS verification code request failed: unknown status code: 17028 Invalid request fields in play_integrity_token

and

>Invoking original failure callbacks after phone verification failure for [My Phone Number], error - This app is not authorized to use Firebase Authentication. Please verify that the correct package name and SHA-1 are configured in the Firebase Console. [ Invalid request fields in play_integrity_token ]

Does anyone has any idea how to resolve this Issue

**Comment by Mr-Hazard** [comment]
i have the smae issue,
if you know the solve please tell me

**Reply by hyderali0889** [reply, marked as answer]
Please do not use the emulator for this purpose. Run your app on a physical device and it will work wonders. Emulators cannot send or receive sms .

**Reply by Mr-Hazard** [reply]
> Please do not use the emulator for this purpose. Run your app on a physical device and it will work wonders. Emulators cannot send or receive sms .
yes iam using physical device but nothing.
i added sha1 and sha2 to my project on firebase.
and i enabled Google Play Integrity Api from google cloud to the project,
and i tried to replace google-service.json file but still nothing change.

**Reply by hyderali0889** [reply]
In my case the problem was the emulator try to turn on the app check in the firebase console and can you show me the error that you receive

**Comment by Davut-Atajanov** [comment]
why does this still not work properly with emulator?
